array.go

```diff
@@ -768,6 +768,97 @@ func (a *Array) NotContainsOnly(values ...interface{}) *Array {
 	return a
 }
 
+// ContainsAny succeeds if array contains at least one element from the given elements.
+// Before comparison, array and all elements are converted
+// to canonical form.
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{"foo", 123, 123})
+//	array.ContainsAny(123, "foo", "FOO")
+//	array.ContainsAny("FOO")
+func (a *Array) ContainsAny(values ...interface{}) *Array {
+	a.chain.enter("ContainsAny()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	elements, ok := canonArray(a.chain, values)
+	if !ok {
+		return a
+	}
+
+	foundAny := false
+
+	for _, expected := range elements {
+		if countElement(a.value, expected) > 0 {
+			foundAny = true
+			break
+		}
+	}
+
+	if !foundAny {
+		a.chain.fail(AssertionFailure{
+			Type:      AssertContainsElement,
+			Actual:    &AssertionValue{a.value},
+			Expected:  &AssertionValue{values},
+			Reference: &AssertionValue{values},
+			Errors: []error{
+				errors.New("expected:" +
+					" array should contain at least one element from reference array"),
+			},
+		})
```

Comment:
When assertion type is AssertContainsElement or AssertNotContainsElement, `Expected` is displayed as `expected element` (according to semantics described in comment to these constants).

Thus, the failure message looks strange: it says that expected element was the whole array. You can see it if you run `go test -v -run TestArrayContainsAny`.

I think we can just remove `Expected` field here and keep only `Actual` and `Reference`.